engine.go

```diff
@@ -30,17 +35,18 @@ func (c *EngineCmd) Run(ctx context.Context, args ...string) error {
 
 	c.close = make(chan struct{})
 
-	// TODO: http server
+	rpcServer := rpc.NewServer()
+	rpcServer.RegisterName("engine", &EngineBackend{})
+	http.Handle("/", &httpServer{rpc: rpcServer})
 
 	go c.RunNode(log)
 
 	return nil
 }
 
 func (c *EngineCmd) RunNode(log *logrus.Logger) {
-
 	log.Info("started")
-
+	http.ListenAndServe(":8090", nil)
```

Comment:
Will change this a bit later, I'd like to close the server etc. in the `Close`, so we can have multiple mock instances and not conflict with global http servers